### PR TITLE
docs: sync English version

### LINKED ISSUE #2583: doc: English version of FAQ
need to sync English version:

https://github.com/apache/apisix/pull/2582

### PR DESCRIPTION
**Pull request by idbeta**
fix #2583

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -297,3 +297,26 @@ By default, APISIX only listens on port 9080 when handling HTTP requests. If you
     ```
 
 2. Reload or restart APISIX
+
+## How does APISIX use etcd to achieve millisecond-level configuration synchronization
+Etcd provides interfaces `wait` and `waitdir` to monitor whether the specified keywords and directories have changed, and return updated data if they have any changes.
```
Comment (membphis) [inline FAQ.md:302]: need a blank line after the title

File: FAQ.md [hunk 2]
```diff
@@ -299,13 +299,14 @@ By default, APISIX only listens on port 9080 when handling HTTP requests. If you
...
 ## How does APISIX use etcd to achieve millisecond-level configuration synchronization
+
 Etcd provides interfaces `wait` and `waitdir` to monitor whether the specified keywords and directories have changed, and return updated data if they have any changes.
 
 Take the `waitdir` interface as an example:
 
 `syntax: res, err = cli:waitdir(dir:string [, modified_index:uint [, timeout:uint] ])`
 
-The `timeout` parameter indicates the connection `timeout` seconds between the calling process and etcd, set `0` to disable timeout.
+The `timeout` parameter indicates the connection `timeout` seconds between the calling process and etcd.
```
Comment (membphis) [inline FAQ.md:309]: Do we need to update the Chinese version.

File: FAQ.md [hunk 3]
```diff
@@ -297,3 +297,27 @@ By default, APISIX only listens on port 9080 when handling HTTP requests. If you
...
+etcd:
+  host:                           # it's possible to define multiple etcd hosts addresses of the same etcd cluster.
+    - "http://127.0.0.1:2379"     # multiple etcd address
+  prefix: "/apisix"               # apisix configurations prefix
+  timeout: 30                     # 30 seconds
+```
+
+APISIX uses the `waitdir` interface to monitor directory changes. The default configuration of timeout is `30` seconds, so APISIX calling process maintains a long connection with etcd for `30` seconds.
+
+If the listening directory is not updated when the APISIX process calls this function, the function returns directly, the long connection is maintained, and the calling process can handle other events. If there is data update within `30` seconds, etcd returns the update result through this function, the calling process handle updated data. If no data returned within `30` seconds, when the timeout period reaches 30 seconds, etcd returns a timeout message through this function, and the calling process handle timeout message, then call the waitdir function to monitor the specified directory again. APISIX realizes real-time configuration updates through the above process.
```
Comment (spacewander) [inline FAQ.md:323]: "the calling process handle" -> "the calling process handles"
"then call" -> "then calls"

File: FAQ.md [hunk 4]
```diff
@@ -297,3 +297,27 @@ By default, APISIX only listens on port 9080 when handling HTTP requests. If you
...
+
+Etcd provides interfaces `wait` and `waitdir` to monitor whether the specified keywords and directories have changed, and return updated data if they have any changes.
+
+Take the `waitdir` interface as an example:
+
+`syntax: res, err = cli:waitdir(dir:string [, modified_index:uint [, timeout:uint] ])`
+
+The `timeout` parameter indicates the connection `timeout` seconds between the calling process and etcd.
+
+APISIX configuration of etcd long connection time is as follows:
```
Comment (moonming) [inline FAQ.md:311]: What is 'Long connection'?

**Review by moonming — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

File: FAQ.md (hunk 4 above)
Comment (moonming) [reply FAQ.md]: any response?

**Comment by membphis** [comment]
@idbeta You’d better reply to the comments when the reviewers leave a comment. This means you read the message and fixed it.

image

image

image

File: FAQ.md (hunk 2 above)
Comment (membphis) [reply FAQ.md:309]: ping @idbeta

File: FAQ.md (hunk 4 above)
Comment (idbeta) [reply FAQ.md]: my apology, already fixed.

File: FAQ.md (hunk 2 above)
Comment (idbeta) [reply FAQ.md:309]: got it

Comment (moonming) [reply FAQ.md:309]: Does this reply make any sense?

File: FAQ.md (hunk 4 above)
Comment (moonming) [reply FAQ.md]: Does this reply make any sense?

File: FAQ.md (hunk 3 above)
Comment (idbeta) [reply FAQ.md]: Ok, I am going to modify this.

File: FAQ.md (hunk 2 above)
Comment (membphis) [reply FAQ.md:309]: > the Chinese version I will fix it later.

do you mean you will fix it in this PR or new PR?

if you want to fix it in new PR, please create an issue about this ^_^

Comment (membphis) [reply FAQ.md:309]: @idbeta any news?

**Review by Miss-you — APPROVED** [review]
LGTM

File: FAQ.md [hunk 5]
```diff
@@ -297,3 +297,27 @@ By default, APISIX only listens on port 9080 when handling HTTP requests. If you
...
+etcd:
+  host:                           # it's possible to define multiple etcd hosts addresses of the same etcd cluster.
+    - "http://127.0.0.1:2379"     # multiple etcd address
+  prefix: "/apisix"               # apisix configurations prefix
+  timeout: 30                     # 30 seconds
+```
+
+APISIX uses the `waitdir` interface to monitor directory changes. The default configuration of timeout is `30` seconds, so APISIX calling process maintains keepalive connection with etcd for `30` seconds.
+
+If the listening directory is not updated when the APISIX process calls this function, the function returns directly, the long connection is maintained, and the calling process can handle other events. If there is data update within `30` seconds, etcd returns the update result through this function, the calling process handle updated data. If no data returned within `30` seconds, when the timeout period reaches 30 seconds, etcd returns a timeout message through this function, and the calling process handles timeout message, then calls the waitdir function to monitor the specified directory again. APISIX realizes real-time configuration updates through the above process.
```
Comment (membphis) [inline FAQ.md:323]: `If the listening directory is not updated when the APISIX process calls this function, the function returns directly`

I think this is wrong. I think you can submit a Chinese version first.

**Review by membphis — CHANGES_REQUESTED** [review]

**Comment by moonming** [comment]
@idbeta any update?

**Comment by membphis** [comment]
I submit a PR to fix the Chinese version: https://github.com/apache/apisix/pull/2844

The Chinese version is wrong.

**Comment by idbeta** [comment]
> I submit a PR to fix the Chinese version: #2844
> 
> The Chinese version is wrong.

thanks, I will update the eng version

**Comment by moonming** [comment]
ping @idbeta

File: FAQ.md (hunk 5 above)
Comment (idbeta) [reply FAQ.md]: fixed.

**Comment by idbeta** [comment]
ping @membphis

**Review by moonming — APPROVED** [review]

**Comment by spacewander** [comment]
@idbeta 
Need to solve conflict.

**Comment by idbeta** [comment]
> @idbeta
> Need to solve conflict.

fixed.

**Review by membphis — APPROVED** [review]

**Pull request merged**